cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityOperator.java

```diff
@@ -109,9 +109,11 @@ protected EntityOperator(Reconciliation reconciliation, HasMetadata resource) {
      * @param reconciliation The reconciliation
      * @param kafkaAssembly desired resource with cluster configuration containing the Entity Operator one
      * @param versions The versions.
+     * @param scramShaAdminApiSupported Indicates whether the SCRAM-SHA API is supported
+     *
      * @return Entity Operator instance, null if not configured in the ConfigMap
      */
-    public static EntityOperator fromCrd(Reconciliation reconciliation, Kafka kafkaAssembly, KafkaVersion.Lookup versions) {
+    public static EntityOperator fromCrd(Reconciliation reconciliation, Kafka kafkaAssembly, KafkaVersion.Lookup versions, boolean scramShaAdminApiSupported) {
```

Comment:
my only concern is about the naming of the new parameter. I know why it's named this way and I can see your point but at same time we pass that parameter this way `!config.featureGates().useKRaftEnabled()`, so we are actually saying if KRaft mode is enabled or not implying that it means SCRAM-SHA supported or not. Can't we imply this inside the EO instead of the API. I mean, having `kraftEnabled` as parameter in the API which translate to not having SCRAM-SHA internally for now. Not a strong opinion, I thought it just worths a discussion :-)

Reply:
I'm not sure it makes any difference one way or another. The question is, where would you wan to change it. The way it is done now, it decides early about the specific of what exactly is not supported. Alternative would be to pass the _KRaft_ flag up to the user operator and decide this only there. So, which way would you prefer?

Reply:
Yeah that is what I would do. Passing to UO if KRaft is enabled or not and encapsulate into the UO logic what is supported or not (i.e. SCRAM-SHA) based on the mode flag.